iptables-module/xt_mpeg2ts.c

```diff
@@ -271,8 +273,9 @@ struct pid_data_t {
 
 /* Data to match a stream / connection */
 struct mpeg2ts_stream_match { /* Like xt_hashlimit: dsthash_dst */
-	__be32 dst_addr; /* MC addr first */
-	__be32 src_addr;
+	int version;
```

Comment:
I see.

I wasn't sure how to determine whether the nf_inet_addr union was currently holding an IPv4 or IPv6 address and therefore added this version variable.

The version variable is pretty much only used to control how the values are printed, ie using NIP6 or %pI4. If there is a NIP46() that could adapt to nf_inet_addr that would significantly reduce the amount of code around print statements.